server/metrics/metrics.go

```diff
@@ -120,6 +152,24 @@ func (m *Metrics) ObserveConnectedUsersTotal(count int64) {
 	}
 }
 
+func (m *Metrics) ObserveChangeEventTotal(changeType string) {
+	if m != nil {
+		m.changeEventTotal.With(prometheus.Labels{"change_type": changeType}).Inc()
+	}
+}
+
+func (m *Metrics) ObserveProcessedChangeEventTotal(changeType string, discardedReason string) {
+	if m != nil {
+		m.processedChangeEventTotal.With(prometheus.Labels{"change_type": changeType, "discarded_reason": discardedReason}).Inc()
+	}
+}
+
+func (m *Metrics) ObserveLifecycleEventTotal(lifecycleEventType string) {
+	if m != nil {
+		m.lifecycleEventTotal.With(prometheus.Labels{"lifecycle_event_type": lifecycleEventType}).Inc()
```

Comment:
nit: is the `lifecycle_` prefix for the `event_type` necessary? (We don't do `change_event_change_type` above.)